docs/presentations/linkedin-meetup-2024-04-24.qmd

```diff
@@ -0,0 +1,725 @@
+penguins = con.table("penguins")
+penguins.group_by("species", "island").agg(count=ibis._.count())
+```
+
+A cloud data platform.
+
+## Oracle {auto-animate="true"}
+
+```python
+import ibis
+
+con = ibis.oracle.connect()
+penguins = con.table("penguins")
+penguins.group_by("species", "island").agg(count=ibis._.count())
+```
+
+A relational database management system.
+
+## Spark {auto-animate="true"}
+
+```python
+import ibis
+
+con = ibis.pyspark.connect()
+penguins = con.table("penguins")
+penguins.group_by("species", "island").agg(count=ibis._.count())
+```
+
+A unified analytics engine for large-scale data processing.
+
+## Trino {auto-animate="true"}
+
+```python
+import ibis
+
+con = ibis.trino.connect()
+penguins = con.table("penguins")
+penguins.group_by("species", "island").agg(count=ibis._.count())
+```
+
+A distributed SQL query engine.
+
+## Flink {auto-animate="true"}
+
+```python
+import ibis
+
+con = ibis.flink.connect()
+penguins = con.table("penguins")
+penguins.group_by("species", "island").agg(count=ibis._.count())
+```
+
+A distributed streaming and batch SQL analytics engine.
+:::
+
+# How does Ibis work?
+
+This is going to be very fast 🏃💨. I will happily answer questions about it 😂.
+
+## Themes
+
+- Immutability
+- Type checking
+- Separation of concerns
+- Extensibility
+- Focus on end-user experience
+- Avoid "common denominator" trap
+- Scale up **and** down
+
+## Components: expressions
+
+:::: {.columns}
+
+::: {.column width="50%"}
+**Expressions**: *interface*
+
+- `StringScalar`, `IntegerColumn`, `Table`
+- `.sum()`, `.split()`, `.join()`
+- No knowledge of specific operation
+:::
+
+::: {.column width="50%"}
+**Operations**: *implementation*
+
+- Specific action: e.g., `StringSplit`
+- Inputs + output dtype, shape
+- Used for compiler dispatch
+:::
+
+::::
+
+::: {.fragment}
+
+### Other goodies
+
+- Type system
+- Pattern matching
+- Graph manipulation/traversal
+
+:::
+
+::: {.fragment .r-fit-text}
+**Goal**: separate API from implementation.
+:::
+
+## Components: expressions
+
+```{python}
+#| echo: true
+#| fig-align: center
+from ibis.expr.visualize import to_graph
+
+expr = (
+    penguins
+    .group_by("species")
+    .agg(avg_bill_mm=_.bill_length_mm.mean())
+)
+
+to_graph(expr)
+```
+
+## Components: compiler
+
+```python
+expr = (
+    penguins
+    .group_by("species")
+    .agg(avg_bill_mm=_.bill_length_mm.mean())
+)
+```
+
+:::: {.r-hstack .semi-fade-out}
+
+::: {.fragment}
+
+```{mermaid}
+graph BT
+  DatabaseTable --> species
+  DatabaseTable --> bill_length_mm
+  bill_length_mm --> Mean
+  species --> Aggregate
+  Mean --> Aggregate
+```
+
+:::
+
+::: {.fragment}
+
+```{mermaid}
+graph BT
+  DatabaseTable2[DatabaseTable] --> species2[species]
+  species2 --> bill_length_mm2[bill_length_mm]
+  bill_length_mm2 --> Mean2[Mean]
+  Mean2 --> Aggregate2[Aggregate]
+```
+
+:::
+
+:::: {.r-stack}
+
+::: {.fragment}
+
+![](./compile.png){width="70%" height="70%" fig-align="center"}
```

Comment:
right here, it is cutoff at the bottom for me -- not a huge deal cuz it's shown on the next page in full

<img width="1666" alt="image" src="https://github.com/ibis-project/ibis/assets/54814569/9e755a46-47be-4052-9b54-dbc6406e1018">

Reply:
yep, leaving it as is